modules/osdk-suggested-namespace-node-selector.adoc

```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-generating-csvs.adoc
+
+:_content-type: PROCEDURE
+[id="osdk-suggested-namespace-default-node_{context}"]
+= Setting a suggested namespace with default node selector
+
+Some Operators expect to run only on primary nodes, which can be done by setting a `nodeSelector` in the Pod spec by the Operator itself.
+
+To avoid getting a duplicated and potential conflicting cluster-wide default `nodeSelector`, you can set a default node selector on the namespace where the Operator runs. The default node selector will take precedence over the cluster default so the cluster default will not be applied to the pods in the Operators namespace.
+
+When adding the Operator to a cluster using OperatorHub, the web console to auto-populates the suggested namespace for the cluster administrator during the installation process, and the suggested namespace created directly using the namespace manifest in YAML included in the cluster service version (CSV).
+
+.Procedure
+
+* In your CSV, set the `operatorframework.io/suggested-namespace-template` annotation to your desired target namespace with namespace default node selector specified:
++
+[source,yaml]
+----
+metadata:
+  annotations:
+    operatorframework.io/suggested-namespace-template: <namespace> <1>
+      {
+        "apiVersion": "v1",
+        "kind": "Namespace",
+        "metadata": {
+          "name": "vertical-pod-autoscaler-suggested-template",
+          "annotations": {
+            "openshift.io/node-selector": ""
+          }
+        }
+      }
```

Comment:
No need "<namespace>" in the example, the full YAML is a template includes example name and the annotation inside for default nodeSelector.

So the wording could be:

_In your CSV, set the `operatorframework.io/suggested-namespace-template` annotation with a manifest for a Namespace object._

_The sample below is a manifest for an example Namespace with namespace default node selector specified:_

```
metadata:
  annotations:
    operatorframework.io/suggested-namespace-template: |-
      {
        "apiVersion": "v1",
        "kind": "Namespace",
        "metadata": {
          "name": "vertical-pod-autoscaler-suggested-template",
          "annotations": {
            "openshift.io/node-selector": ""
          }
        }
      }
```